### ISSUE TITLE
new windows build

### ISSUE THREAD

**Issue by kofifus** [nickbnf/glogg#243]
The latest windows build on https://glogg.bonnefon.org/ is 1.1.4 from 18 May 2017

When can a new version build be expected ? is there anywhere to get windows binaries of latest code ?

Thanks!

**Comment by variar** [comment]
Latest master build artifacts for windows can be downloaded from https://ci.appveyor.com/project/nickbnf/glogg/builds/21795819/artifacts

**Comment by kofifus** [comment]
thanks! the windows installer fails with this msg:

![image](https://user-images.githubusercontent.com/3926071/51574156-c52c8100-1f00-11e9-922c-f2ed9c6502c4.png)

**Comment by variar** [comment]
For some reason this file is not included in installer.  I'll look into it. Meanwhile you can download zip-archive and copy missing file to glogg install folder.

**Comment by kofifus** [comment]
OK great thanks I got it to work - another file that is missing in the installer is `libgcc_s_dw2-1.dll`

Also, glogg help->about shows this:

![image](https://user-images.githubusercontent.com/3926071/51640689-4eeb5580-1fb8-11e9-9fdb-215863e50c04.png)

It is not clear what version number is this matching - can that be changed to the correct version ?

**Comment by variar** [comment]
Thanks for testing. This is built from the latest sources. The long line of characters in about window is just git identifier for the latest commit that has got into this build. Meaningful version will appear when new official release is created.

**Comment by rob-X1** [comment]
Note that the build is 32bit only. If your target platform is 64bit only (e.g. a WinRE 64bit system) this build is of not use to you.

**Comment by variar** [comment]
I think this issue should stay open till official stable release, so link to the build can be easily found.

Appveyor does not provide Qt version for mingw 64 bit. And code changes are required to build with msvc compiler, so building 64-bit variant automatically is not possible at the moment. 

My fork should include recent changes from glogg and has 32/64 builds, however 19.01 versions are experimental. Hope to release stable version in February.

**Comment by NewUserHa** [comment]
It's February now.
hope there'll be a new windows version soon

**Comment by NewUserHa** [comment]
@variar I tried your forked klogg. it works. thanks for all you guys' work
can I suggest a feature here?: 
  detecting encoding then apply ASAP instead finishing a whole file would be great

**Comment by variar** [comment]
@NewUserHa, can you describe the issue that suggested feature shoul fix?

Both glogg and klogg detect encoding based on first several bytes of file and use it for loading rest of the file. Glogg uses custom encoding detection and klogg uses Qt for UTF-8/16/32 and uchardet for some other 8-bit encodings. Both algorithms are not perfect but should work well enough.

**Comment by NewUserHa** [comment]
klogg opened a text file, >>3GB and >300k lines and gb2312 encoding.
all Chinese characters are garbled texts until the whole indexing of file finished completely.

**Comment by variar** [comment]
Indeed, encoding for displaying text is set after indexing is finished. I'll try to set correct encoding after it is detected.

**Comment by kofifus** [comment]
May now so two years (!!) since last official windows build - is glogg dead ?

**Comment by variar** [comment]
There has been some commits in 2018. I can prepare a build from latest sources if someone is interested. 

@nickbnf I can try to make a PR that will add artifacts from Travis/Appveyor builds to a github release on each commit, but this will require some configuration to add proper repository access keys.

**Comment by kofifus** [comment]
+1

**Comment by kofifus** [comment]
@variar any news on this ?

**Comment by variar** [comment]
Been a little busy releasing new stable version of klogg.

I've updated pull request to build glogg from latest sources using Qt 5.12

builds from https://ci.appveyor.com/project/nickbnf/glogg/builds/27640336
[glogg-1.1.4.90_x64-setup.zip](https://github.com/nickbnf/glogg/files/3649508/glogg-1.1.4.90_x64-setup.zip)
[glogg-1.1.4.90_x64.zip](https://github.com/nickbnf/glogg/files/3649509/glogg-1.1.4.90_x64.zip)
[glogg-1.1.4.90_x86-setup.zip](https://github.com/nickbnf/glogg/files/3649510/glogg-1.1.4.90_x86-setup.zip)
[glogg-1.1.4.90_x86.zip](https://github.com/nickbnf/glogg/files/3649511/glogg-1.1.4.90_x86.zip)

**Comment by thirdwheel** [comment]
@variar, the libstdc++ DLL file is missing from the 64-bit installer.  Had to copy it from the non-setup ZIP file.

**Comment by variar** [comment]
I've updated pull request to build glogg from latest sources using Qt 5.12.5

builds from https://ci.appveyor.com/project/nickbnf/glogg/builds/30172532
[glogg-1.1.4.93_x64-setup.zip](https://github.com/nickbnf/glogg/files/4074315/glogg-1.1.4.93_x64-setup.zip)
[glogg-1.1.4.93_x64.zip](https://github.com/nickbnf/glogg/files/4074316/glogg-1.1.4.93_x64.zip)
[glogg-1.1.4.93_x86-setup.zip](https://github.com/nickbnf/glogg/files/4074317/glogg-1.1.4.93_x86-setup.zip)
[glogg-1.1.4.93_x86.zip](https://github.com/nickbnf/glogg/files/4074318/glogg-1.1.4.93_x86.zip)

**Comment by kofifus** [comment]
Thanks variar !

I'm wondering why keep both glogg and klogg project ? I ended up switching to the build here https://github.com/variar/klogg/releases/tag/continuous-win

**Comment by variar** [comment]
> I'm wondering why keep both glogg and klogg project ? I ended up switching to the build here https://github.com/variar/klogg/releases/tag/continuous-win

Personally I'm happy to hear this, thanks @kofifus :) However, glogg has well established reputation and many users. Sometimes having a well-known stable tool that does it's job well for your use-cases is ok, no need to change anything. Not everybody need new features, but having small annoying bugs fixed can make a day little better.

I'm actually thinking of creating a separate fork that will have no new features but provide bugfix and maintenance (like updating Qt version).  I wonder if anybody watching this repository is interested in such idea?

**Comment by kofifus** [comment]
Is there anywhere a list of new features in klogg ?

**Comment by variar** [comment]
I've released [klogg 20.12](https://github.com/variar/klogg/releases/tag/v20.12).

Here is a list of new features compared with current glogg 1.1.4 (accumulated over several years of development):

- Faster initial file indexing. On ssd drives glogg is limited by cpu and does not utilize full IO speed of the drive.
- Faster searching. Klogg uses several threads to perform regex matching and has several other optimizations.
- Better file encoding detection and more encodings available. Glogg uses its own implementation for detecting text encoding, and klogg uses well-known uchardet library.
- Better support for log colorization. In klogg it is possible to configure several sets of rules to do text highlighting and switch between on the fly. Highlighters can be set to colorize only matched text inside the line.
- Better support for `tail` mode. Klogg uses hashing to check if changes appeared at the end of file or in the middle. This reduces file reloads and prevents subtle bugs. Also klogg uses ESFW library to detect changes in files. It proved to provide more accurate events across all supported platforms.
- Better styling support. Klogg detects dark OS themes and switches color palette. There is an option to select different Qt style base on system-wide installed Qt style plugins. And there is separate dark style based on QDarkStyleSheet.
- Automatic unpacking of some archives to temp directory using KArchive library from KDE Frameworks.
- A separate scratchpad window to save temporary text and do basic transformations (indent json, decode base64 etc.)
- New favorites menu to save files that are opened not frequently enough to stay in recent files history.
- An option to download log file from some URL and open it.
- An option to clear log file.
- An option to save part of log file or search results.
- An option to restrict searching to some part of huge file.
- More options for tabs (opening file location, copying file path to clipboard, tab renaming).
- Experimental multi-window mode. Allows to open several klogg windows.
- A portable version for Windows (x86 and x64).
- An appimage version for Linux along with deb/rpm packages (deb and rpm repositories provided by Bintray).
- A DMG package for MacOS with more intuitive installation guide and digitally signed to make Gatekeeper happy.
- Precomiled klogg packages use latest Qt 5.15 version on Windows and MacOS that should have better HIDPI displays support and overall less bugs.

Of course "better" can be biased as I am the author of klogg. I would appreciate any independent comparison of glogg and klogg features.
